test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/Streaming/ODataJsonLightStreamWritingTests.cs

```diff
@@ -0,0 +1,712 @@
+                writer.WriteEnd(); // resource
+            },
+            expectedPayload);
+        }
+
+        [Fact]
+        public void CannotWriteValueForODataProperty()
+        {
+            Action writeWithExtraValue = () => RunTest((ODataWriter writer) =>
+            {
+                // write a binary property using a stream
+                writer.WriteStart(resource);
+                writer.WriteStart(new ODataProperty
+                {
+                    Name = "textAsStream",
+                    Value = new ODataPrimitiveValue("text")
+                });
+                writer.WritePrimitive(new ODataPrimitiveValue("text"));
+                writer.WriteEnd(); // property
+                writer.WriteEnd(); // resource
+            },
+            null);
+
+            writeWithExtraValue.ShouldThrow<ODataException>().WithMessage(Strings.ODataWriterCore_PropertyValueAlreadyWritten("textAsStream"));
+        }
+
+        [Fact]
+        public void CannotStreamValueForODataProperty()
+        {
+            Action writeWithExtraValue = () => RunTest((ODataWriter writer) =>
+            {
+                // write a binary property using a stream
+                writer.WriteStart(resource);
+                writer.WriteStart(new ODataProperty
+                {
+                    Name = "textAsStream",
+                    Value = new ODataPrimitiveValue("text")
+                });
+                using (TextWriter textWriter = writer.CreateTextWriter())
+                {
+                    textWriter.Write("text");
+                    textWriter.Flush();
+                };
+                writer.WriteEnd(); // property
+                writer.WriteEnd(); // resource
+            },
+            null);
+
+            writeWithExtraValue.ShouldThrow<ODataException>().WithMessage(Strings.ODataWriterCore_PropertyValueAlreadyWritten("textAsStream"));
+        }
+
+        [Fact]
+        public void CannotWriteValueTwice()
+        {
+            Action writeWithExtraValue = () => RunTest((ODataWriter writer) =>
+            {
+                // write a binary property using a stream
+                writer.WriteStart(resource);
+                writer.WriteStart(new ODataPropertyInfo
+                {
+                    Name = "textAsStream"
+                });
+                writer.WritePrimitive(new ODataPrimitiveValue("text"));
+                using (TextWriter textWriter = writer.CreateTextWriter())
+                {
+                    textWriter.Write("text");
+                    textWriter.Flush();
+                };
+                writer.WriteEnd(); // property
+                writer.WriteEnd(); // resource
+            },
+            null);
+
+            writeWithExtraValue.ShouldThrow<ODataException>().WithMessage(Strings.ODataWriterCore_PropertyValueAlreadyWritten("textAsStream"));
+        }
+
+        [Fact]
+        public void CannotWriteAndStreamValue()

```

Comment:
>CannotWriteAndStreamValue [](start = 20, length = 25)

This test case looks doing different things?

Reply:
Good catch. I renamed this as "CannotWriteValueTwice" and added two new tests: "CannotWriteAndStreamProperty" and "CannotStreamAndWriteProperty" (which try to write the same property twice, once as stream and once as property, in the order indicated).

---
In reply to: [268365098](https://github.com/OData/odata.net/pull/1204#discussion_r268365098) [](ancestors = 268365098)